openpgp/packet/private_key.go

```diff
@@ -104,26 +119,31 @@ func NewSignerPrivateKey(creationTime time.Time, signer crypto.Signer) *PrivateK
 		pk.PublicKey = *NewECDSAPublicKey(creationTime, pubkey)
 	case ecdsa.PublicKey:
 		pk.PublicKey = *NewECDSAPublicKey(creationTime, &pubkey)
-	case ed25519.PublicKey:
+	case *ed25519.PublicKey:
```

Comment:
Why are we accepting both objects and pointers?

Reply:
There was a comment above saying:
```
	// In general, the public Keys should be used as pointers. We still
	// type-switch on the values, for backwards-compatibility.
```
So - for RSA and ECDSA, it used to be values only, but they switched to pointers.
For EdDSA, this PR also switches from values to pointers for consistency.
I tried removing the non-pointer cases, and it broke some tests - but maybe we could remove them anyway in a later PR and fix the tests, if we're OK with the (minor) backwards incompatibility.